modules/thermo-cycler/production/firmware_uploader.py

```diff
@@ -32,11 +33,14 @@ def build_arg_parser():
                             help='Firmware file (default: ..production/firmware/thermo-cycler-arduino.ino.bin)')
     return arg_parser
 
-def find_opentrons_port():
+def find_opentrons_port(bootloader=False):
     retries = 5
     while retries:
         for p in comports():
             if p.vid == OPENTRONS_VID:
+                if bootloader:
+                    if p.pid != TC_BOOTLOADER_PID:
```

Comment:
Shouldn't we also skip this if !bootloader and p.pid == TC_BOOTLOADER_PID

Reply:
Not really. Case in point- right after flashing the bootloader, the board has bootloader PID only. When we have to upload firmware on it, the script starts by looking for an opentrons port so that it can perform the 1200 baud touch and put it in bootloader mode. Right now, it performs the 1200baud touch on bootloader port itself (which has no effect) and then sees the bootloader port (which was already there) to upload firmware to. If we made it so that if !bootloader it should only look for a non bootloader port then it will never be able to find such a port after a bootloader upload.
If we want to make the conditions tighter so that this script doesn't work on anything other than a thermocycler, then we can have it look for ports that are either tc_bootloader or tc port only